scripts/zones/West_Ronfaure/npcs/Vilatroire.lua

```diff
@@ -5,29 +5,200 @@
+                    player:updateEvent(3)
+                end
+            elseif questIntermediateTeamwork == QUEST_ACCEPTED then
+                -- https://www.bg-wiki.com/bg/Intermediate_Teamwork
+                if partySameRaceCount == partySizeRequirement then
+                    -- race requirements met
+                    player:setLocalVar("intermedTmwrk_pass", 1)
+                    player:updateEvent(15, 2)
+                else
+                    -- not the same race
+                    player:updateEvent(4)
+                end
+            elseif questAdvancedTeamwork == QUEST_ACCEPTED then
+                -- https://www.bg-wiki.com/bg/Advanced_Teamwork
+                if partySameJobCount == partySameJobCount then
+                    -- race requirements met
+                    player:setLocalVar("advTmwrk_pass", 1)
+                    player:updateEvent(15, 3)
+                else
+                    -- not the same job
+                    player:updateEvent(5)
+                end
+            end
+        else
+            -- need more party members
+            player:updateEvent(1)
+        end
+    end
 end
 
 function onEventFinish(player, csid, option)
-end
+    -- csid 129 is the event for when they have selected ready/not ready
+    
+    if csid == 129 and option == 0 then
+        local questIntroToTeamwork = player:getQuestStatus(SANDORIA, tpz.quest.id.sandoria.INTRODUCTION_TO_TEAMWORK);
+        local questIntermediateTeamwork = player:getQuestStatus(SANDORIA, tpz.quest.id.sandoria.INTERMEDIATE_TEAMWORK);
+        local questAdvancedTeamwork = player:getQuestStatus(SANDORIA, tpz.quest.id.sandoria.ADVANCED_TEAMWORK);
+
+        if questIntroToTeamwork == QUEST_ACCEPTED and player:getLocalVar("introToTmwrk_pass") == 1 then
+            -- check their inventory
+            npcUtil.completeQuest(player, SANDORIA, tpz.quest.id.sandoria.INTRODUCTION_TO_TEAMWORK, {
+                item = 13442,
+                fame = 80, -- fame defaults to 30 if not set
+                title = tpz.title.THIRDRATE_ORGANIZER,
+            })
+        elseif questIntermediateTeamwork == QUEST_ACCEPTED and player:getLocalVar("intermedTmwrk_pass") == 1 then
+             -- check their inventory
+            npcUtil.completeQuest(player, SANDORIA, tpz.quest.id.sandoria.INTERMEDIATE_TEAMWORK, {
+                item = 4994,
+                fame = 80, -- fame defaults to 30 if not set
+                title = tpz.title.SECONDRATE_ORGANIZER,
+            })
+        elseif questAdvancedTeamwork == QUEST_ACCEPTED and player:getLocalVar("advTmwrk_pass") == 1 then
+             -- check their inventory
+            npcUtil.completeQuest(player, SANDORIA, tpz.quest.id.sandoria.ADVANCED_TEAMWORK, {
+                item = 13459,
+                fame = 80, -- fame defaults to 30 if not set
+                title = tpz.title.FIRSTRATE_ORGANIZER,
+            })
+    end 
+    elseif csid == 131 and option == 1 then
```

Comment:
```lua
end
elseif
```
Will probably make Lua unhappy!

Reply:
Close, just need 4 more spaces on line 193.

Reply:
New to lua, is there a reason why it will make it unhappy? :)

Reply:
It won't, but the indentation is off on line 193 which caused some confusion.